src/MBP_UserImport_Source_AfterSchool.php

```diff
@@ -19,16 +19,83 @@ class MBP_UserImport_Source_AfterSchool extends MBP_UserImport_BaseSource
   /**
    * Supported key / columns in CSV file from source.
    */
-  private function setKeys() {
+  protected function setKeys() {
 
     $keys = [
-      '???',
+      'SentToPhone',
+      'SenderName',
+      'ReceiverName',
+      'SchoolID',
+      'SchoolName',
+      'SchoolShort',
+      'SchoolAbbreviation',
+      'Message'
     ];
 
     return $keys;
   }
   
   /**
+   * Logic to determine if data from row in After School CSV file can be processed.
+   *
+   * @param array $data
+   *   Values being processed from row of data in CSV file.
+   */
+  public function canProcess($data) {
+
+    if (empty($data['SentToPhone'])) {
+      return false;
+    }
+
+    // Validate phone number based on the North American Numbering Plan
+    // https://en.wikipedia.org/wiki/North_American_Numbering_Plan
+    $regex = "/^(\d[\s-]?)?[\(\[\s-]{0,2}?\d{3}[\)\]\s-]{0,2}?\d{3}[\s-]?\d{4}$/i";
```

Comment:
regexp is a bit of a "black art" to me. I don't pretend to understand all of the related syntax. The example you're pointing to does not seem to support xxx-xxx-xxxx. The "-" character is often used in the Americas where AfterSchool numbers originate.

I'm going to stick with the current pattern but that you for researching.